platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBook.java

```diff
@@ -552,7 +560,7 @@ public void deserialize(@NonNull final SerializableDataInputStream in, final int
         if (version < ClassVersion.ADDRESS_BOOK_STORE_SUPPORT) {
             round = UNKNOWN_ROUND;
             if (!orderedNodeIds.isEmpty()) {
-                nextNodeId = new NodeId(orderedNodeIds.get(getSize() - 1).id() + 1);
+                nextNodeId = orderedNodeIds.get(getSize() - 1).getOffset(1);
```

Comment:
shouldnt we also serialize and deserialize nextNodeId?

Reply:
This code path is under the branch condition `if (version < ClassVersion.ADDRESS_BOOK_STORE_SUPPORT) {`.  I preserved the behavior.  I assume that the `nextNodeId` field was not introduced until after the AddressBook Store Support.   Later in the method the next node id is deserialized. 
```
        round = in.readLong();
        if (version < ClassVersion.SELF_SERIALIZABLE_NODE_ID) {
            nextNodeId = new NodeId(in.readLong());
        } else {
            nextNodeId = in.readSerializable(false, NodeId::new);
        }
```